packages/yew/src/virtual_dom/mod.rs

```diff
@@ -34,9 +34,78 @@ pub use self::vnode::VNode;
 pub use self::vtag::VTag;
 #[doc(inline)]
 pub use self::vtext::VText;
+use std::fmt::Formatter;
+use std::ops::Deref;
+use std::rc::Rc;
 
 /// Attribute value
-pub type AttrValue = Cow<'static, str>;
+#[derive(Eq, PartialEq, Debug)]
+pub enum AttrValue {
+    /// String living for `'static`
+    Static(&'static str),
+    /// Owned string
+    Owned(String),
+    /// Reference counted string
+    ReferenceCounted(Rc<str>),
+}
+
+impl Deref for AttrValue {
+    type Target = str;
+
+    fn deref(&self) -> &Self::Target {
+        match self {
+            AttrValue::Static(s) => *s,
+            AttrValue::Owned(s) => s.as_str(),
+            AttrValue::ReferenceCounted(s) => &*s,
+        }
+    }
+}
+
+impl From<&'static str> for AttrValue {
+    fn from(s: &'static str) -> Self {
+        AttrValue::Static(s)
+    }
+}
+
+impl From<String> for AttrValue {
+    fn from(s: String) -> Self {
+        AttrValue::Owned(s)
+    }
+}
+
+impl From<Rc<str>> for AttrValue {
+    fn from(s: Rc<str>) -> Self {
+        AttrValue::ReferenceCounted(s)
+    }
+}
+
+impl Clone for AttrValue {
+    fn clone(&self) -> Self {
+        match self {
+            AttrValue::Static(s) => AttrValue::Static(s),
+            AttrValue::Owned(s) => AttrValue::Owned(s.clone()),
+            AttrValue::ReferenceCounted(s) => AttrValue::ReferenceCounted(Rc::clone(s)),
+        }
+    }
+}
+
+impl PartialEq<String> for AttrValue {
```

Comment:
We should implement `PartialEq<str>` too. Maybe we don't even need `PartialEq<String>` then, but not quite sure about that.

Reply:
How about removing `PartialEq` implementations altogether? `AsRef` and `Deref` can get the job done.

I added the `String` one because some place needed it but seems like it works fine if I use `as_ref`.

Reply:
I removed the implementation. Let me know if one should be added.